url10fb6bd43e_guwere_P2PSearch_tgz-pJ8-MainJ8/src/Main.java

```diff
@@ -0,0 +1,134 @@
+import java.net.*;
+import java.util.Random;
+/** CS4032 Distributed Systems Project 2013, Trinity College Dublin
+ * Created by tony on 08/12/13.
+ * This program licences under the MIT license
+ * You can see a copy of the license here http://opensource.org/licenses/MIT
+ *
+ * @author Antonio Nikolov
+ */
+public class Main {
+    public static void main(String [] args){
+        if(args.length == 0){
+            System.out.println("Starting simulation.");
+            initialiseSimulation();
+        }else if(args[0].equals("--boot"))
+            initialiseAsJoining(args);
+        else if(args[0].equals("--bootstrap"))
+            initialiseAsBootstrap(args);
+        else{
+            System.out.println("Please specify --boot <numeric_id> or --bootstrap <ip> <numeric_id>");
+        }
+    }
+    private static void initialiseAsJoining(String [] args){
+        try {
+            int port = NetworkProperties.nPort;
+            NetworkProperties.nAddress = InetAddress.getLocalHost().getHostAddress();  //should give local ip
+            InetSocketAddress bootstrapAddress = new InetSocketAddress(args[1],port);
+            DatagramSocket socket = new DatagramSocket(port);
+            PeerSearchSimplified node1 = new Node();
+            node1.init(socket);
+            node1.joinNetwork(bootstrapAddress,"myname",args[2]);//i.e bootstrap node
+
+        } catch (SocketException e) {
+            System.err.println("Could not open socket.");
+            e.printStackTrace();
+        } catch (UnknownHostException e) {
+            e.printStackTrace();
+        }
+
+
+    }
+    private static void initialiseAsBootstrap(String [] args) {
+        try {
+            int port = NetworkProperties.nPort;
+            NetworkProperties.nAddress = InetAddress.getLocalHost().getHostAddress();  //should give local ip
+            InetSocketAddress bootstrapAddress = new InetSocketAddress("127.0.0.1",port);
+            DatagramSocket socket = new DatagramSocket(port);
+            PeerSearchSimplified node1 = new Node();
+            node1.init(socket);
+            node1.joinNetwork(bootstrapAddress,"myname","myname");//i.e bootstrap node. won' try to connect
+
+        } catch (SocketException e) {
+            System.err.println("Could not open socket.");
+            e.printStackTrace();
+        } catch (UnknownHostException e) {
+            e.printStackTrace();
+        }
+    }
+    private static void initialiseSimulation(){
+        int port = NetworkProperties.nPort;
+        String address = NetworkProperties.nAddress;
+        InetSocketAddress bootstrapAddress = new InetSocketAddress(address,port);
+        DatagramSocket socket = null;
+        try {
+            socket = new DatagramSocket(port);
```

Comment:
Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

**Problem**
This line of code might contain a resource leak. Resource leaks can cause your system to slow down or crash.

**Fix**
Consider closing the following resource: *socket*. Execution paths that do not contain closure statements were detected. Ensure *socket* is closed in a try-finally block or declared in a try-with-resources block.

**More info**
[View resource management guidelines at oracle.com](https://www.oracle.com/technetwork/articles/java/trywithresources-401775.html) (external link).